backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/descriptors/ClassLayoutBuilder.kt

```diff
@@ -278,7 +278,7 @@ internal class ClassLayoutBuilder(val irClass: IrClass, val context: Context) {
         val superVtableEntries = if (irClass.isSpecialClassWithNoSupertypes()) {
             emptyList()
         } else {
-            val superClass = irClass.getSuperClassNotAny() ?: context.ir.symbols.any.owner
+            val superClass = irClass.getRuntimeSuperClassNotAny() ?: context.ir.symbols.any.owner
```

Comment:
Looks unpredictable and thus harmful. Is this actually required?

Reply:
On the one hand, it’s indeed looks kinda unclear. On the other, there are several places that require uniform approach. I’m not sure, is there a better approach that doesn’t require a lot of refactoring.

Reply:
The thing is: the list of these places is not well-defined.
This partucular place isn't supposed to work properly with `superClass` different from actual super class.
Generally type info for Kotlin subclass of Obj-C class is a hack, and it should continue being local. Patching only one (unused) field of actual TypeInfo is local enough. Hacking different non-trivial fragments of compiler code is not so local.